### ISSUE TITLE
Exression tan(x)*cot(x) is not constant

### ISSUE THREAD

**Issue by rpisarev** [sympy/sympy#10702]
In [11]: a = tan(x)*cot(x)

In [12]: a.is_constant()
Out[12]: True

That's wrong

**Comment by thisch** [comment]
Why do you think that's wrong?

**Comment by rpisarev** [comment]
Because:

In [1]: from sympy import *

In [2]: x = Symbol('x', real=True)

In [3]: a = tan(x)*cot(x)

In [4]: a.is_constant()
Out[4]: True

In [5]: simplify(a)
Out[5]: 1

In [6]: a.subs(x, 0)
Out[6]: nan

Domain of a isn't S.Reals

**Comment by argriffing** [comment]
> In [11]: a = tan(x)*cot(x)
> 
> In [12]: a.is_constant()
> Out[12]: True
> 
> That's wrong

Sympy is not WolframAlpha, but WolframAlpha says this is 1.

**Comment by thisch** [comment]
I agree with @argriffing , per definition tan(x)*cot(x) has to be equal to 1. 

I think the problem is that ``.is_constant`` 'sees' that ``tan`` is the inverse of ``cot`` and therefore returns ``True``, but ``.subs`` does not see it.

```
from sympy import *
x = Symbol('x', real=True)
a = tan(x)*cot(x)
b = simplify(a)
print(a.subs(x, 0))
print(b.subs(x, 0)
```

returns:
nan
1

**Comment by rpisarev** [comment]
Hm, yes, WolframAlpha says this is 1. It's very strange. E.g. for (x-1)/(x-1)  WolframAlpha says 1  (for x!=1) 

I want to see  tan(x)*cot(x) 1 (for x != pi*k/2, k in Z)

**Comment by thisch** [comment]
So ``simplify`` should then return 1 only if it knows that x != pi * k / 2, k in Z. ``is_constant`` in this case should probably return ``None``  @asmeurer.

**Comment by thisch** [comment]
Thinking a bit more about this issue, I guess that WA is correct that cot(x)*tan(x) is 1 for all x. Try to write cot and tan in terms of sin and cos and take the limit as x->0. It follows from L'Hospital's Rule that cot(x)*tan(x) is 1 at x=0.  The same is true for (1-x)/(1-x).

**Comment by jksuom** [comment]
It appears not explicitly stated in the docs but I think that an expression is usually considered constant when it has the same value everywhere in its domain of definition.

**Comment by rpisarev** [comment]
Yes, it is 1 for all x in domain this expression. But domain is not whole space (R or C). That is why I think that is not constant. I will try discovery #6054 for solving.

**Comment by ankyAS** [comment]
I think @rpisarev is correct as the tan and cot are discontinous function and the value of tanx is not defined at (pi/2)*k.So it should not always be 1.
@thisch  writing tan(x) in terms of  sin(x) and cos(x) will led to value being undefined at (pi/2) while both sin(x) and cos(x) are defined at pi/2
